hyperon_das_atomdb/adapters/ram_only.py

```diff
@@ -138,26 +183,59 @@ def _build_targets_list(self, link: Dict[str, Any]):
             count += 1
         return targets
 
-    def _update_index(self, atom: Dict[str, Any]):
+    def _update_index(self, atom: Dict[str, Any], **kwargs):
         atom_type = atom['named_type']
-        self._add_atom_type(_name=atom_type)
-        if 'name' not in atom:
-            handle = atom['_id']
+        if kwargs.get('delete_atom', False):
+            if atom is None:
+                raise LinkDoesNotExist('link not exist')
+
+            link_handle = atom['_id']
+
+            named_type_exists = any(
```

Comment:
The same comments I made for node type deletion applies here. But here it's even worst as you are iterating on all links (which are usually in larger number). Please delete this check.